cvmfs/catalog_mgr_rw.h

```diff
@@ -58,16 +58,25 @@ class Statistics;
 }
 
 namespace catalog {
+template <class CatalogMgrT>
+class CatalogBalancer;
+}
+
+namespace catalog {
 
 class WritableCatalogManager : public SimpleCatalogManager {
  public:
+  friend class CatalogBalancer<WritableCatalogManager>;
```

Comment:
You need to grant the CatalogBalancer<WritableCatalogManager> permission to merge catalogs and read the thresholds